lib/jxl/enc_modular.cc

```diff
@@ -1230,12 +1246,10 @@ Status ModularFrameEncoder::PrepareStreamParams(const Rect& rect,
       Channel gc(r.xsize(), r.ysize());
       gc.hshift = fc.hshift;
       gc.vshift = fc.vshift;
+      JXL_ASSERT((r.x0() & 0xf) == 0);
       for (size_t y = 0; y < r.ysize(); ++y) {
-        const pixel_type* const JXL_RESTRICT row_in = r.ConstRow(fc.plane, y);
-        pixel_type* const JXL_RESTRICT row_out = gc.Row(y);
-        for (size_t x = 0; x < r.xsize(); ++x) {
-          row_out[x] = row_in[x];
-        }
+        memcpy(gc.Row(y), JXL_ASSUME_ALIGNED(r.ConstRow(fc.plane, y), 64),
```

Comment:
This assume_aligned looks like it will break if we ever make pixel_type a 16-bit type.

Do we really need it? Does it make a significant difference?

Reply:
Nah it can be dropped. I just saw a lot of unaligned memmoves in perf top, but probably only a few of those originate here (adding the assume aligned doesn't seem to significantly reduce the time spent in unaligned memmoving). And the assumption seems to be wrong even for int32_t pixel_type, so I'll drop it.

Reply:
Dropped it. Didn't make any difference, and was wrong too. Also, I hope in the future we can just avoid this copying, either by letting local modular images be rects within the global one, or by skipping the global modular image when the only global transforms are RCTs, and converting the input buffer directly to local modular images.